### PR TITLE
refactor(grid): use logical CSS properties

### PR DESCRIPTION
**Pull request by simeonoff**
Closes #  

### Additional information (check all that apply):
 - [ ] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [ ] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

**Review by didimmova — CHANGES_REQUESTED** [review]
instead of including the if-rtl and if-ltr mixins, you should use [dir='ltr'] and [dir='rtl'] selectors

File: projects/igniteui-angular/src/lib/core/styles/components/grid/_excel-filtering-theme.scss [hunk 1]
```diff
@@ -282,8 +285,8 @@
         igx-list {
             flex-grow: 1;
             overflow: hidden;
-            margin-left: -#{rem(16px)};
-            margin-right: -#{rem(16px)};
+            margin-inline-start: -#{rem(16px)};
```
Comment (didimmova) [inline projects/igniteui-angular/src/lib/core/styles/components/grid/_excel-filtering-theme.scss:288]: you can write just margin-inline: -#{rem(16px)}; and it'll cover both properties

File: projects/igniteui-angular/src/lib/core/styles/components/grid/_excel-filtering-theme.scss [hunk 2]
```diff
@@ -222,19 +220,24 @@
...
 
-                span,
-                igx-icon {
-                    margin: initial;
-                }
+    %grid-excel-sort .igx-button-group {
+        [dir='rtl'] & {
+            flex-direction: row-reverse;
 
-                @include icon-w-margin(rem(8px), $right);
```
Comment (didimmova) [inline projects/igniteui-angular/src/lib/core/styles/components/grid/_excel-filtering-theme.scss:237]: Something here became wrong and the excel-filtering doesn't look the same as before.

This is how it looked like before:
Screenshot 2022-04-13 at 14 17 41


This is how it looks like now: 
Screenshot 2022-04-13 at 14 17 34

File: projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-theme.scss [hunk 3]
```diff
@@ -2339,43 +2336,43 @@
             }
         }
 
-        @include if-rtl() {
+        [dir='rlt'] & {
```
Comment (didimmova) [inline projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-theme.scss:2339]:
```suggestion
        [dir='rtl'] & {
```

File: projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-theme.scss [hunk 4]
```diff
@@ -2516,8 +2513,8 @@
         display: flex;
         align-items: center;
         justify-content: center;
-        padding-#{$right}: 8px;
-        margin-#{$left}: 8px;
+        padding-inline-end: 8px;
+        margin-inline-start: 8px;
```
Comment (didimmova) [inline projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-theme.scss:2517]: Maybe we can use the `rem()` function here?

File: projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-theme.scss [hunk 5]
```diff
@@ -2678,11 +2675,11 @@
 
         &::after {
             @extend %filtering-scroll-mask;
-            #{$left}: calc(100% + 6px);
-            background: linear-gradient(to #{$right}, var-get($theme, 'filtering-row-background'), transparent);
+            inset-inline-start: calc(100% + 6px);
+            background: linear-gradient(to end, var-get($theme, 'filtering-row-background'), transparent);
```
Comment (didimmova) [inline projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-theme.scss:2679]: You said this one doesn't work this way?

File: projects/igniteui-angular/src/lib/core/styles/components/grid/_excel-filtering-theme.scss (hunk 2 above)
Comment (simeonoff) [reply projects/igniteui-angular/src/lib/core/styles/components/grid/_excel-filtering-theme.scss:237]: Correct, now it looks like it should.

File: projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-theme.scss (hunk 5 above)
Comment (simeonoff) [reply projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-theme.scss]: Correct, this has been resolved in the latest commit.

**Review by didimmova — APPROVED** [review]

**Pull request merged**